crypto/ui/ui_util.c

```diff
@@ -109,7 +104,8 @@ static int ui_read(UI *ui, UI_STRING *uis)
             char result[PEM_BUFSIZE];
             const struct pem_password_cb_data *data =
                 UI_method_get_ex_data(UI_get_method(ui),
-                                      ui_method_data_index());
+                                      RUN_ONCE(&get_index_once,
+                                               ui_method_data_index));
```

Comment:
they return a status code.  Returning a value isn't common.  a file-static with the index is cleaner.   IMO.  If you disagree, I won't fight it.